Client/Frontend/Settings/SearchSettingsTableViewController.swift

```diff
@@ -177,14 +226,18 @@ class SearchSettingsTableViewController: ThemedTableViewController, FeatureFlagg
     }
 
     override func numberOfSections(in tableView: UITableView) -> Int {
-        guard featureFlags.isFeatureEnabled(.feltPrivacySimplifiedUI, checking: .buildOnly) else {
-            return Section.allCases.count - 1
+        sectionsToDisplay = [Section.defaultEngine, Section.quickEngines]
```

Comment:
I learned that, apparently, UIKit can [call `numberOfSections` multiple times](https://stackoverflow.com/questions/2638359/uitableview-delegate-method-called-twice/4986682).

That's totally alright in this case, since the array is so small, and the feature flags won't change while the view is on the screen—but a fun tidbit to remember for the future about how table views get rendered.

Reply:
i did realize this when i was debugging and stepping through my breakpoints, which is why i was originally uncertain that it was the best way to go about it, but if i recall correctly, when i tried to implement the logic outside of numberOfSections, things got a little weird